ui/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -180,6 +188,10 @@ const mapStateToProps = (state, ownProps) => {
     fromAddress,
   );
 
+  const isFailedTransaction = Boolean(
+    transactionsToDisplayOnFailure[fullTxData.id],
```

Comment:
The presence of the transaction in this collection does not indicate that it is a failed transaction. We should be checking here that the transaction _isn't_ in `unapprovedTxs` - not that it _is_ in `transactionsToDisplayOnFailure`.

Reply:
@Gudahtt this doesn't work as expected. At this point in the execution, the transaction is in unapprovedTxs even though it's a failed tx.

Reply:
Hmmm. That should not be the case. The list of `unapprovedTxs` necessarily have a status of "unapproved".

Reply:
When I use the above condition (isFailedTransaction = isn't in unapprovedTxs), I can't go past the confirm page because the confirm button is not shown (i.e isFailedTransaction is true).

Pretty late for me. I will take a closer look in the morning

Reply:
Interesting! Sounds good, I can take a look tomorrow as well

Reply:
Ok, so I fixed this. The real issue was fullTxData.id is number and unapproveTxs ids are string.

Reply:
This still looks like the opposite of what it should be. Did you forget a `!`?

Reply:
Did you discover what was going on here after?